enclave-runtime/src/lib.rs

```diff
@@ -332,22 +346,30 @@ pub unsafe extern "C" fn init_parentchain_components(
 ) -> sgx_status_t {
 	info!("Initializing light client!");
 
+	match init_parentchain_params_internal(params, params_size, latest_header, latest_header_size) {
+		Ok(()) => sgx_status_t::SGX_SUCCESS,
+		Err(e) => e.into(),
+	}
+}
+
+unsafe fn init_parentchain_params_internal(
```

Comment:
As this is not a `C` function, just `unsafe Rust`, could you add a `Safety` docs explaining the expectations for the input parameters a caller needs to check? 

I am also surprised clippy did not complain as there is a lint for this:
https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc
Maybe because the function is not `pub`?

Reply:
Yeah. this in only because it is not `pub`. I agree, but I think I actually solved this in a better way than just documenting 9844f8f6918851e2210b38026eec50b837c958bb. :)

Thanks for the nitpick ;)